artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerSenderContext.java

```diff
@@ -733,20 +734,31 @@ private static boolean hasCapabilities(Symbol symbol, Source source) {
       return false;
    }
 
-   private static String createQueueName(String clientId,
+   private static String createQueueName(boolean useLegacyQueueNaming,
+                                         String clientId,
                                          String pubId,
                                          boolean shared,
                                          boolean global,
                                          boolean isVolatile) {
-      String queue = clientId == null || clientId.isEmpty() || global ? pubId : clientId + "." + pubId;
-      if (shared) {
-         if (queue.contains("|")) {
-            queue = queue.split("\\|")[0];
-         }
-         if (isVolatile) {
-            queue = "nonDurable" + "." + queue;
+      if (useLegacyQueueNaming) {
+         String queue = clientId == null || clientId.isEmpty() ? pubId : clientId + "." + pubId;
```

Comment:
Ok. I think we should instead leave the fix (checking the global flag and not adding the container-id to the queue name) in place to still resolve the original bug. Its that or raising another PR to put it back in after taking it back out in this one.